raw_data_QA/check_raw_case_death_data.py

```diff
@@ -0,0 +1,658 @@
+        # could add start and end date here Natasha
+        args.states = latest_df["state"].unique()
+
+    # Iterate thru states
+    for var in variables:
+        (
+            z_avg_list,
+            z_latest_avg_list,
+            days_over_thres_list,
+            states_list,
+            rmse_new_list,
+            rmse_latest_list,
+        ) = ([], [], [], [], [], [])
+        for state in args.states:
+            # Grab Data To Compare from CAN Data Caches
+            this_prod_df = get_state(prod_df, state, args)
+            this_latest_df = get_state(latest_df, state, args)
+
+            # Aggregate Datasets (i.e. combine counties to state level and calculate new cases and deaths)
+            prod_ag = aggregate_df(this_prod_df, args)
+            latest_ag = aggregate_df(this_latest_df, args)
+            (
+                avg_z,
+                latest_avg_z,
+                days_over_z,
+                rmse_latest,
+                abnormal_old,
+                abnormal_new,
+                average_new_p_diff,
+            ) = compare_data(var, prod_ag, latest_ag, prod_name, latest_name, args, state,)
+
+            if abnormal_old:
+                z_avg_list.append(avg_z)
+                z_latest_avg_list.append(latest_avg_z)
+                days_over_thres_list.append(days_over_z)
+                states_list.append(state)
+                rmse_latest_list.append(rmse_latest)
+                historical_report_string = (
+                    state
+                    + "'s historical "
+                    + var
+                    + " data disagree (RMSE: "
+                    + str(rmse_latest)
+                    + ")\n"
+                )
+                output_report.write(historical_report_string)
+                _logger.warning(historical_report_string)
+                sentry_sdk.capture_message(historical_report_string)
+            if abnormal_new:
+                new_data_report_string = (
+                    state
+                    + "'s "
+                    + "latest "
+                    + var
+                    + " is on average "
+                    + str(average_new_p_diff)
+                    + "% different relative to past data.\n"
+                )
+                output_report.write(new_data_report_string)
+                _logger.warning(new_data_report_string)
+                sentry_sdk.capture_message(new_data_report_string)
+
+        # Create meta-compare charts for all abnormal areas
+        states_list = list(dict.fromkeys(states_list))
+        make_meta_comparison_plot(
+            states_list,
+            days_over_thres_list,
+            "days_over_thres",
+            rmse_latest_list,
+            "rmse_latest",
+            z_avg_list,
+            "z_avg",
+            z_latest_avg_list,
+            "z_latest",
+            var,
+        )
+
+    output_report.close()
+    os.system("rm -rf " + args.output_dir + "/" + args.output_data_dir)
```

Comment:
please don't merge with this line that can so easily become `rm -fr /` :-O

I'd like to disentangle your code from changes on disk as much as possible. It looks like you are using output_dir for temporary storage. Instead use https://docs.python.org/3/library/tempfile.html#tempfile-examples
```
# create a temporary directory using the context manager
>>> with tempfile.TemporaryDirectory() as tmpdirname:
...     print('created temporary directory', tmpdirname)
```

Reply:
fixed in latest version. Will see if the build system likes the switch

Reply:
I still see `os.system("rm -rf "...` which isn't needed because you used tempfile with a context manager so it cleans up for you. Is there a newer version?

Reply:
yep fixed